src/components/transactions/Supply/SupplyModalContent.tsx

```diff
@@ -41,209 +42,228 @@ export enum ErrorType {
-    ((!user.isInIsolationMode && !poolReserve.isIsolated) ||
-      (user.isInIsolationMode &&
-        user.isolatedReserve?.underlyingAsset === poolReserve.underlyingAsset))
-  ) {
-    healthFactorAfterDeposit = calculateHealthFactorFromBalancesBigUnits({
-      collateralBalanceMarketReferenceCurrency: totalCollateralMarketReferenceCurrencyAfter,
-      borrowBalanceMarketReferenceCurrency: valueToBigNumber(
-        user.totalBorrowsMarketReferenceCurrency
-      ),
-      currentLiquidationThreshold: liquidationThresholdAfter,
-    });
-  }
+    poolReserve,
+    userReserve,
+    isWrongNetwork,
+    nativeBalance,
+    tokenBalance,
+  }: ModalWrapperProps) => {
+    const { marketReferencePriceInUsd, user } = useAppDataContext();
+    const { currentMarketData, currentNetworkConfig } = useProtocolDataContext();
+    const { mainTxState: supplyTxState, gasLimit, txError } = useModalContext();
+    const { supplyCap, debtCeiling } = useAssetCaps();
+    const {
+      poolComputed: { minRemainingBaseTokenBalance },
+    } = useRootStore();
 
-  // ************** Warnings **********
-  // isolation warning
-  const hasDifferentCollateral = user.userReservesData.find(
-    (reserve) => reserve.usageAsCollateralEnabledOnUser && reserve.reserve.id !== poolReserve.id
-  );
-  const showIsolationWarning: boolean =
-    !user.isInIsolationMode &&
-    poolReserve.isIsolated &&
-    !hasDifferentCollateral &&
-    (userReserve && userReserve.underlyingBalance !== '0'
-      ? userReserve.usageAsCollateralEnabledOnUser
-      : true);
-
-  // TODO: check if calc is correct to see if cap reached
-  const capReached =
-    poolReserve.supplyCap !== '0' &&
-    valueToBigNumber(amount).gt(
-      new BigNumber(poolReserve.supplyCap).minus(poolReserve.totalLiquidity)
+    // states
+    const [amount, setAmount] = useState('');
+    const [maxSelected, setMaxSelected] = useState(false);
+    const amountRef = useRef<string>();
+    const supplyUnWrapped = underlyingAsset.toLowerCase() === API_ETH_MOCK_ADDRESS.toLowerCase();
+
+    const walletBalance = supplyUnWrapped ? nativeBalance : tokenBalance;
+
+    const supplyApy = poolReserve.supplyAPY;
+
+    // Calculate max amount to supply
+    const maxAmountToSupply = getMaxAmountAvailableToSupply(
+      walletBalance,
+      poolReserve,
+      underlyingAsset,
+      minRemainingBaseTokenBalance
     );
 
-  // handle error for supply cap reached
-  let blockingError: ErrorType | undefined = undefined;
-  if (!supplyTxState.success) {
-    if (capReached) {
-      blockingError = ErrorType.CAP_REACHED;
+    const handleChange = (value: string) => {
+      const isMaxSelected = value === '-1';
+      const inputValue = maxSelected ? maxAmountToSupply.toString(10) : value;
+      const decimalTruncatedValue = roundToTokenDecimals(inputValue, poolReserve.decimals);
+      amountRef.current = decimalTruncatedValue;
+      setAmount(decimalTruncatedValue);
+      setMaxSelected(isMaxSelected);
```

Comment:
I think adding this as state is causing a weird bug with the asset input. If you type in `1` for the value, and then hit the max button, you can replicate the issue. It won't actually add in the max value in that case.

It doesn't look `maxSelected` needs to be state, I think we can just compute it on each render based on the amount, unless I'm missing something. But there also might be a bug with the asset input that's causing this issue.